src/vm/rejit.cpp

```diff
@@ -718,16 +961,31 @@ HRESULT ReJitManager::BindILVersion(
         // twice in a row, without us having a chance to jmp-stamp the code yet OR
         // while iterating through instantiations of a generic, the iterator found
         // duplicate entries for the same instantiation.)
+        // TODO: this assert likely needs to be removed. This code path should be
+        // hit for any duplicates, and that can happen regardless of whether this
+        // is the first ReJIT or not.
         _ASSERTE(ilCodeVersion.HasDefaultIL());
 
         *pILCodeVersion = ilCodeVersion;
+
+        if (!fSuppressGetParams)
+        {
+            // We want to handle the suppression of GetReJITParamters so that it handles duplicates correctly.
+            // There could be a case where the method that a profiler requested ReJIT on also ends up in the 
+            // inlining graph from a different method. In that case we should override the previous setting,
+            // but we should never override a request to get the callback with a request to suppress it.
+            pILCodeVersion->SetSuppressReJITCallback(false);
```

Comment:
Nit: Naming this SetEnableReJITCallback and flipping the meaning of the flag would avoid needing to reason about double negatives that occur here.